### ISSUE TITLE
exuse me !!!!do you have some tricks when training model?why do i train MLT dataset as default parameters and after 50000iters,it can detect nothing?

### ISSUE THREAD

**Issue by cjt222** [eragonruan/text-detection-ctpn#29]
exuse me !!!! do you have some tricks when training model?why do i train MLT dataset as default parameters and after 50000 iters,it can detect nothing?

**Comment by eragonruan** [comment]
@cjt222 the model I provided is trained directly by the parameters I set in text.yml. There are many reason why no result is saved, you can first use tensorboard to check the log, see whether the model has converged or not? Also, If the testing image is too large, you should resize it before testing, resize its short side to 600pix is recommended

**Comment by cjt222** [comment]
when i train model , if image is too large, do i need to resize image unitl short size is 600pix? @eragonruan

**Comment by cjt222** [comment]
i train voc data you provided and add maybe 500 images like below,image is really large,maybe 3000x2200.  
i set your default parameters as batch size 300 and lr 0.0001,and train 200000 iters,but nothing can be detected ,loss is about 0.9, i do not know if it did not converged
![00045_00035](https://user-images.githubusercontent.com/17508662/32584564-e11ee6fa-c533-11e7-8ee5-3675e909b099.jpg)

**Comment by eragonruan** [comment]
@cjt222 about the first question, I have done when prepare the training data, you can find in [prepare_training_data/split_label.py#L27(https://github.com/eragonruan/text-detection-ctpn/blob/3abf200be51fb577682681a85814aceb2460f804/prepare_training_data/split_label.py#L27)

about your second question, the loss function used for classification is cross entropy, 0.9 did not converge at all, 0.69 is the threshold. in my experiment, the classification can reach to 0.15

**Comment by eragonruan** [comment]
I have try the image you provided, the model I provided you handle it well.
![test](https://user-images.githubusercontent.com/11270892/32584870-90d0f2cc-c535-11e7-95f2-5ad40e95b22b.jpg)

**Comment by cjt222** [comment]
![image](https://user-images.githubusercontent.com/17508662/32591836-0b99bf9a-c55c-11e7-8c64-0949944d5701.png)
hello, i saw some layers of VGG_train.py and VGG_test.py  will not change parameters, what is your idea?

**Comment by cjt222** [comment]
@eragonruan

**Comment by eragonruan** [comment]
@cjt222 This is used to accelerate the training process since the model is initialized with VGG pretrained in ImageNet, so I assumed that they have been trained well.   If you want, you can unfix these layers for better performance

**Comment by cjt222** [comment]
in face,  whatever data, e.g VOC dataset or my dataset, it did not converge anymore and it vibrated from 0.2 to 0.9, and even i test image in train datasets, but it can detect nothing.  my lr is 0.0001, i thought if i set lr 0.000001.,model can learn nothing.i do not where i am wrong @eragonruan

**Comment by cjt222** [comment]
hi, maybe split_label.py has some error, it can not convert my label correctly

**Comment by eragonruan** [comment]
@cjt222 Are you using the latest code?

**Comment by cjt222** [comment]
yes,thanks for your answer,do you meet this problem:when i test some images,if shape of image is same, the boundingbox is all same, i just train 15000 iters, i am not sure if the reason is  model not converge @eragonruan

**Comment by eragonruan** [comment]
@cjt222 no, I haven't met this problem, you can use tensorboard to checkout whether the model has converged or not. It seems that tyour model did not converge

**Comment by JeonghanSeock** [comment]
I have also experienced this problem. (no image file in result folder.)
How did you set the checkpoint when you run demo.py after training?
trainging -> edit text.yml -> run demo.py -> check result folder.
edit text.yml line35 to #checkpoints_path: checkpoints/
edit text.yml lint36 to checkpoints_path: output/ctpn_end2end/voc_2007_trainval
I hope it will be of help.

**Comment by xiaomaxiao** [comment]
Zoom in the parameter MAX_HORIZONTAL_GAP. 50->60 or bigger.

**Comment by eragonruan** [comment]
Thanks for the update, close for now
